### PR TITLE
chore: update readme [NONE]

### PR DESCRIPTION
**Pull request by marcolink**
## Summary

Update all *.md files with a new header (stolen from [here](https://github.com/contentful/contentful.js/tree/beta-v10))

- Introduce a TOC for README.md 
- Remove old unused example files

[Rendered version](https://github.com/contentful/contentful-management.js/tree/chore/update-readme)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,13 +92,6 @@ Other browsers should also work, but at the moment we're only running automated
 
 To get started with the Contentful Management JS library you'll need to install it, and then get credentials which will allow you to access your content in Contentful.
 
-- [Installation](#installation)
```
Comment (andreascful) [inline README.md:39]: Why were those links removed?

File: README.md [hunk 2]
```diff
@@ -1,19 +1,75 @@
...
 [![gzip bundle size](http://img.badgesize.io/https://unpkg.com/contentful-management/dist/contentful-management.browser.min.js?compression=gzip)](https://unpkg.com/contentful-management/dist/contentful-management.browser.min.js)
 
+**What is Contentful?**
+
 [Contentful](https://www.contentful.com) provides a content infrastructure for digital teams to power content in websites, apps, and devices. Unlike a CMS, Contentful was built to integrate with the modern software stack. It offers a central hub for structured content, powerful management and delivery APIs, and a customizable web app that enable developers and content creators to ship digital products faster.
 
+<details>
+<summary>Table of contents</summary>
+
+<!-- TOC -->
```
Comment (andipaetzold) [inline README.md:44]: Why a TOC here? GitHub automatically creates an outline on when visiting the repo.

File: README.md [hunk 3]
```diff
@@ -1,19 +1,75 @@
...
+
+- [Features](#features)
+- [Supported Environments](#supported-environments)
+- [Getting Started](#getting-started)
+  - [Installation](#installation)
+    - [Node](#node-)
```
Comment (andipaetzold) [inline README.md:50]: nit: There is a `-` after `#node` and `#browser`

**Review by andipaetzold — COMMENTED** [review]
* It's "JavaScript", not "Javascript"
* The dots between Migration, changelog and contributing are missing

File: README.md (hunk 2 above)
Comment (marcolink) [reply README.md:44]: @andipaetzold the readme is also rendered on npmjs.com

File: README.md (hunk 1 above)
Comment (marcolink) [reply README.md]: Good catch - re-added!

**Comment by marcolink** [comment]
> * It's "JavaScript", not "Javascript"
> * The dots between Migration, changelog and contributing are missing

Thank @andipaetzold - fixed!

File: README.md (hunk 3 above)
Comment (marcolink) [reply README.md:50]: @andipaetzold without them, the linking is not working.

**Review by andipaetzold — APPROVED** [review]

**Pull request merged**